### ISSUE TITLE
[TF2] [MvM] Starting to heal a player as they die instantly revives them

### ISSUE THREAD

**Issue by powerlord** [ValveSoftware/Source-1-Games#1574]
If you start healing a player with a Medigun as they die, they instantly revive.

This isn't a server setting as it happens on Valve's Mann Up servers.

For reference, my Medic loadout when this happens is:

* Strange Syringe Gun
* The Kritzkrieg
* Strange Ubersaw
* The Surgeon's Stahlhelm
* Physician's Procedure Mask 
* The Surgeon's Stethoscope
* Battery Canteens

**Comment by powerlord** [comment]
Only took two and a half years and thanking someone else for reporting this, but this was finally fixed.

**Comment by sigsegv-mvm** [comment]
> Only took two and a half years and thanking someone else for reporting this, but this was finally fixed.

@powerlord Sorry they didn't thank you... you certainly were first to report this. So you definitely deserve to have gotten some credit.

If it helps, I can give you some context about what/when I reported to Valve. Here's the email I sent them on February 2 2016:
[Email gist](https://gist.github.com/sigsegv-mvm/7895ed546524d0dddd6c7929d8106ca3)
(Please forgive the lack-of-text-wrapping in that gist.)

And here's a video from January 30 where I used dtwatchent to demonstrate that the max health of the reanimator wasn't getting set immediately:
https://www.youtube.com/watch?v=3mi0QUzISDY

So anyway, I guess they probably didn't bother to read your bug report, but they did read my email, for whatever reason. (And then they waited until like 5 months after getting the email to actually do the fix. Because Valve.)

It's certainly frustrating, getting Valve to read and act on bug reports. I've basically come to the conclusion that the most effective way to get them to actually take action is to essentially do all the work for them. I almost always try to send emails where, instead of saying "here's the bug, fix it", I'll instead do a bunch of work to investigate it myself, and then email them with "here's the bug, here's the exact cause that I tracked down, and here's the exact thing you should change in the code to fix it". It's kind of sad that I feel the need to do that, but it does seem to get things fixed more often. (And even then they still manage to fsck up the fixes sometimes.)

For the last year, I'd been mostly emailing my bug reports to the "TF Team" address through the Valve contact form on their website, and it usually worked, but a lot of stuff just seemed to get ignored or slip through the cracks. But relatively recently, Eric Smith just flat out emailed me and basically said I should just directly email bug reports to him and Dave Riller, to ensure that they see them and that they get directed to the proper people to fix them. So that's what I'm doing now, and it does seem to be a more reliable way of getting things acted on rather than ignored. I even get replies to my emails on a somewhat regular basis, which I didn't before.

Anyway, I hope that helps explain things a bit.

**Comment by powerlord** [comment]
@sigsegv-mvm I've found in the past that unless I manage to catch the attention of someone at Valve, my bug reports go largely ignored.

For example, I've managed to catch John Schoenick's attention with my bug reports for the Map Workshop when it first launched in beta last year.  In fact, nearly every update in June and July 2015 had some bug fixed that I'd reported... none mention my name anywhere.

Heck, the [August 18th 2015](https://wiki.teamfortress.com/wiki/August_18,_2015_Patch) Community Request was done by **my** request so that I could use it in map voting plugins, including those shipped with SourceMod 1.8.

**Comment by sigsegv-mvm** [comment]
@powerlord I only started getting name mentions in the update notes around the past month or so, I think; before that, I think the only one was for the exploit I reported (and they always do name mentions for those). I'm pretty sure that me getting name mentions recently has to do with the fact that I send my reports directly to Eric Smith's email now (Eric seems to like me).

Perhaps John Schoenick just doesn't do name mentions when he adds to the patch notes?

I know that in the past, when I submitted everything through the "TF Team" contact form, I'd frequently find that they did fix bugs that I'd reported, but I'd almost never get email replies (not even "fixed, thanks"), and the fix would typically not even show up in the patch notes at all, let alone with a name drop. (I was pretty annoyed that I literally had to open a disassembler, technically violating the Steam terms of service, just to find out whether or not they had actually fixed a bug.)

So I think it depends a lot on which particular guy on the team receives your report and via what mechanism.
